tests/test_factories.py

```
@@ -0,0 +1,341 @@
+            # Hier kommt kein LazyFunction, sondern direkt das factory-Ergebnis
+            self.assertIs(result, dummy)
+
+    def test_fk_with_factory_existing_instance(self):
+        dummy1 = DummyForeignKey(name="a")
+        dummy2 = DummyForeignKey(name="b")
+        field = DummyModel._meta.get_field("dummy_fk")
+
+        class GMC:
+            pass
+
+        GMC.Factory = lambda **kwargs: dummy  # type: ignore
```

Ruff (0.11.9)

Undefined name `dummy`

(F821)